### ISSUE TITLE
Web interface stops responding

### ISSUE THREAD

**Issue by 10481** [mholgatem/ThermOS#1]
I've been playing around with the idea of using this, however, I've found that the web interface religiously stops working. I _think_ it might be related to leaving it up in a browser session that goes idle, but I'm not positive. I do know that I can interact with it for a long period of time, but if I let my computer go to sleep or my phone sleep, depending on where I was last checking it, it won't respond when I return to it or open from another device. As soon as I restart the thermostat_web service, it carries on as normal. 

No changes have been made to the interface. It's on a fresh install of raspbian and only has the temp sensor connected as per the instructions.

**Comment by mholgatem** [comment]
yeah, it tends to be a problem with serving the app up with Flask rather than a traditional web server engine. I think that I might use Gunicorn to act as a go between. I'll leave this issue open until I have time to delve into this a bit more.

**Comment by mholgatem** [comment]
I have integrated gunicorn into the project to help serve the flask application. It seems to be working well, but I need to test the update script a little bit more.

**Comment by 10481** [comment]
It's working well now for me. I've ran it through the usual steps that guaranteed it stopping for me prior and it keeps on going. This is very cool.

Here's my `update.sh` output

```
Downloading/unpacking gunicorn
  Downloading gunicorn-19.6.0-py2.py3-none-any.whl (114kB): 114kB downloaded
Installing collected packages: gunicorn
Compiling /tmp/pip-build-dNZDbZ/gunicorn/gunicorn/workers/_gaiohttp.py ...
  File "/tmp/pip-build-dNZDbZ/gunicorn/gunicorn/workers/_gaiohttp.py", line 84
    yield from self.wsgi.close()
             ^
SyntaxError: invalid syntax

Successfully installed gunicorn
Cleaning up...
Creating backup in: /home/pi/thermos_backup/2016_09_29-12_02_15/
M	logs/logs.db
M	logs/thermostat.db
Already on 'master'
Your branch is up-to-date with 'origin/master'.
Saved working directory and index state WIP on master: 7fbd39f Update README.md
HEAD is now at 7fbd39f Update README.md
remote: Counting objects: 54, done.
remote: Total 54 (delta 32), reused 32 (delta 32), pack-reused 22
Unpacking objects: 100% (54/54), done.
From https://github.com/mholgatem/ThermOS
   7fbd39f..75b1e9a  master     -> origin/master
Updating 7fbd39f..75b1e9a
error: The following untracked working tree files would be overwritten by merge:
	update.sh
Please move or remove them before you can merge.
Aborting
On branch master
Your branch is behind 'origin/master' by 16 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   logs/logs.db
	modified:   logs/thermostat.db

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	update.sh

no changes added to commit (use "git add" and/or "git commit -a")
Dropped refs/stash@{0} (a631eb3755b937352ffe956ed4412ec7de2759f7)
Warning: Unit file of thermostat-web.service changed on disk, 'systemctl daemon-reload' recommended.
Warning: Unit file of thermostat-daemon.service changed on disk, 'systemctl daemon-reload' recommended.
Reloading daemon and restarting services...
```

**Comment by mholgatem** [comment]
cool. I'm glad that it worked well for you. There's this one error that I want to try and fix where if you edit part of a file and I edit the same part of the file on my end, then it messes up. But other than that it seems to be working well!

**Comment by 10481** [comment]
This is going really well. I haven't hooked it up in place of the thermostat yet, but it has been running non-stop just fine now. I've also added a button for lights that uses your popup and houses my RFoutlet interface for controlling lights. Now that I've got that done, I think I'll style it appropriately and add in the garage doors as well. I'd also like to take the current day's logs from the SprinklersPi that I also have running and maybe display those times in the bar down below with the time, but that might be out of my scope. Not sure yet. This is a perfect little hub for controlling everything. 

Anyway, I'll leave it up to you to close this when you see fit, as it's originally tied to the interface crashing. I think you've fixed that in my opinion.
